WebUI/Pages/AdminDashboard/AdminGames.razor

```diff
@@ -1,5 +1,163 @@
+<br />
+
+@* ---Game Form--- *@
+@if (showForm)
+{
+    <EditForm Model="@adminGameViewModel" OnValidSubmit="@HandleFormSubmit">
+        <br />
+        <DataAnnotationsValidator />
+        <ValidationSummary />
+        <br />
+        <h3>Edit Game</h3>
+        <br />
+        <table>
+            <tr>
+                <td>Name:</td>
+                <td><InputText id="Game_Name_Edit" @bind-Value="adminGameViewModel.Name" /></td>
+            </tr>
+            <tr>
+                <td>Description:</td>
+                <td><InputText id="Game_Description_Edit" @bind-Value="adminGameViewModel.Description" /></td>
+            </tr>
+            <tr>
+                <td>Price:</td>
+                <td><InputNumber id="Game_Price_Edit" @bind-Value="adminGameViewModel.Price" /></td>
+            </tr>
+            <tr>
+                <td>Genre:</td>
+                <td>
+                    <InputSelect id="Game_Genre_Edit" @bind-Value="adminGameViewModel.GenreName">
+                        <option value="">None</option>
+                        @foreach (var genre in gameGenres)
+                            {
+                            <option value="@genre.Name">@genre.Name</option>
+                            }
+                    </InputSelect>
+                </td>
+            </tr>
+        </table>
+        <br />
+        <button class="btn btn-primary" type="submit">Save</button>
+        <button class="btn btn-secondary" @onclick="@ToggleGameForm">Cancel</button>
+    </EditForm>
+}
+@* --- Display Game List --- *@
+else if (games != null && games.Count != 0)
+{
+    <div class="row" style="padding-top: 10px">
+        @foreach (var game in games)
+        {
+            <div class="col-sm-6">
+                <div class="card">
+                    <div class="card-body">
+                        <h5 class="card-title">@game.Name</h5>
+                        <p class="card-text">@game.Description</p>
+                        <button @onclick="() => InitEdit(game)" class="btn btn-primary">Edit Game</button>
+                        <button @onclick="() => HandleDelete(game)" class="btn btn-danger">Delete Game</button>
+                    </div>
+                </div>
+            </div>
+        }
+    </div>
+}
+else
+{
+    <br />
+    <h3>No games available... create a new game!</h3>
+}
+
+
+@code {
+
+    #region Declarations
+    private Game selectedGame = new Game();
+    private GameVM adminGameViewModel = new GameVM();
+    private List<Game> games;
+    private List<GameGenre> gameGenres;
+    private bool showForm = false;
+    private bool isAdd = true;
+    #endregion
+
+    //Initializes game list and game genre list
+    protected override async Task OnInitializedAsync()
+    {
+        await FetchGames();
+        gameGenres = await gameGenreService.GetGameGenres();
+    }
+
+    private async Task FetchGames()
+    {
+        games = await gameService.GetGames();
+    }
```

Comment:
In this case, I would suggest inlining this method in all the places it's used, cause it's only a one-liner, so extracting it into a separate method doesn't bring much value.